src/System.Threading.Channels/src/System/Threading/Channels/AsyncOperation.cs

```diff
@@ -93,14 +150,20 @@ public T GetResult(short token)
             }
 
             ExceptionDispatchInfo error = _error;
-            T result = _result;
+            TResult result = _result;
             _currentId++;
-            _state = (int)States.Released; // only after fetching all needed data
+
+            if (_pooled)
+            {
+                Volatile.Write(ref _continuation, s_availableSentinel); // only after fetching all needed data
```

Comment:
>_continuation [](start = 35, length = 13)

wouldn't _continuation already equals to s_availableSentinel as teh constructor setting that?

Reply:
never mind, I am seeing you reset it in TryOwnAndReset

---
In reply to: [172628879](https://github.com/dotnet/corefx/pull/27780#discussion_r172628879) [](ancestors = 172628879)

Reply:
> wouldn't _continuation already equals to s_availableSentinel as teh constructor setting that?

In order to use a pooled instance, you need to call TryOwnAndReset, which transitions _continuation from s_availableSentinel to null.